### PR TITLE
[Merged by Bors] - feat(Order/SuccPred/TypeTags): `SuccOrder (Multiplicative X)` and friends

### PR DESCRIPTION
**Pull request by pechersky**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/TypeTags.lean
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/TypeTags.lean:1]: Could this become `Algebra.Order.SuccPred.TypeTags`?

**Review by YaelDillies — COMMENTED** [review]
Can you prove the simp lemmas of the form `succ (ofMul _) = ofMul (succ _)`, `succ (toMul _) = toMul (succ _)`, etc?

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**